apps/meteor/app/livechat/server/lib/Contacts.js

```diff
@@ -1,14 +1,27 @@
 import { check } from 'meteor/check';
+import { Meteor } from 'meteor/meteor';
 import s from 'underscore.string';
 
 import { LivechatVisitors, LivechatCustomField, LivechatRooms, Rooms, LivechatInquiry, Subscriptions } from '../../../models';
+import { Users } from '../../../models/server/raw';
 
 export const Contacts = {
 	registerContact({ token, name, email, phone, username, customFields = {}, contactManager = {} } = {}) {
 		check(token, String);
 
 		const visitorEmail = s.trim(email).toLowerCase();
 
+		if (contactManager?.username) {
+			// verify if the user exists with this username and has a livechat-agent role
+			const user = Promise.await(Users.findOneByUsername(contactManager.username, { projection: { roles: 1 } }));
+			if (!user) {
+				throw new Meteor.Error('error-contact-manager-not-found', `No user found with username ${contactManager.username}`);
+			}
+			if (!user.roles || !Array.isArray(user.roles) || !user.roles.includes('livechat-agent')) {
```

Comment:
Personally i don't like checking for roles here, but for now that would work.

Pls create a task for changing this check to be a permission in the future 👀

Reply:
I agree to tackle this issue separately since we also have similar usage elsewhere in the code like https://github.com/RocketChat/Rocket.Chat/blob/993745d3996450036e8c39150a0b6926cdbbd869/apps/meteor/app/livechat/server/api/lib/users.js#L51

Here's the task for it btw: https://app.clickup.com/t/2f82np6